src/core/internal/cppdemangle.d

```diff
@@ -0,0 +1,268 @@
+/**
+* This module provides functions to demangle C++ symbols
+*
+* For Posix platform the demangling is done by function `__cxa_demangle` presents in Itanium C++ ABI.
+* For Windows platform the demangling is done by function `UnDecorateSymbolName` presents in the Debug Help Library
+* If the function is not found or something fails, the original mangled C++ name will be returned.
+*
+* Copyright: Copyright © 2020, The D Language Foundation
+* License: $(HTTP www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+* Authors: Ernesto Castellotti
+* Source: $(DRUNTIMESRC core/internal/_cppdemangle.d)
+*/
+module core.internal.cppdemangle;
+
+version (Windows)
+{
+    version (ANSI) {} else version = Unicode;
+}
+
+/**
+* Demangles C++ mangled names.
+*
+* If it is not a C++ mangled name or cppdemangle is not supported by your platform, the original mangled C++ name will be returned.
+* The optional destination buffer and return will be contains the same string if the demangle is successful.
+*
+ * Params:
+ *  buf = The string to demangle.
+ *  dst = The destination buffer, if the size of the destination buffer is <= the size of cppdemangle output the return string would be incomplete.
+ *  withPrefix = If true, add the prefix (followed by a space) before the C++ demangled name, make sure the target buffer is at least large to contain the prefix.
+ *  prefix = Specifies the prefix to be added to the beginning of the string containing the demangled name
+ *
+ * Returns:
+ *  The demangled name or the original string if the name is not a mangled C++ name.
+ */
+char[] cppdemangle(const(char)[] buf, char[] dst, bool withPrefix = false, string prefix = "[C++]") @safe
```

Comment:
So do you think it's better to provide a fixed size buffer like I did with the Windows version?

Reply:
Yeah, I'd say only accept non-null buffer for the time being

Reply:
it's a problem because __cxa_demangle was created to call realloc automatically if the buffer is too small, I have no way of preventing this.
For this reason I passed null to the function which returned me a string pointer of unpredictable size, then I copy the characters in the buffer passed to the function up to the available size in the buffer.

Reply:
Oh right, read the documentation again and it's not a great behavior. Especially the reliance on `malloc`. Reading again, it *must* be allocated with malloc, which makes the API very unsafe for external user. Can you document this ?

Reply:
you are really unsafe, I think the only way is to let it allocate then copy the result (if the buffer has enough space) and free the memory